chat-plugins/room-events.js

```diff
@@ -9,61 +9,87 @@
 
 'use strict';
 
+const fs = require('fs');
+const path = require('path');
+
+const EVENTS_FILE = path.resolve(__dirname, '../config/chat-plugins/events.json');
+const ALLOWED_HTML = ['a', 'font', 'i', 'u', 'b', 'strong', 'em', 'small', 'sub', 'sup', 'ins', 'del', 'code'];
+
+let events = {};
+try {
+	events = require(EVENTS_FILE);
+} catch (e) {
+	if (e.code !== 'MODULE_NOT_FOUND') throw e;
+}
+if (!events || typeof events !== 'object') events = {};
+
+function saveEvents() {
+	fs.writeFileSync(EVENTS_FILE, JSON.stringify(events));
+}
+
+
 exports.commands = {
 	events: 'roomevents',
 	roomevent: 'roomevents',
 	roomevents: {
 		'': function (target, room, user) {
-			if (room.battle) return this.errorReply("This command is not meant for battle rooms.");
-			if (!room.events || Object.keys(room.events).length === 0) {
+			if (!room.chatRoomData) return this.errorReply("This command is unavailable in temporary rooms.");
```

Comment:
They aren't, because it gets passed through ``this.canHTML`` which filters out specific HTML depending on the user and room, i.e., images in groupchats.